src/test/java/ejercicio/banco/repository/MySqlAccountRepositoryTest.java

```diff
@@ -33,31 +33,32 @@
     public void testFindAllReturnsFullList() throws SQLException {
         executeQuery(util, "select * from ACCOUNTS");
 
-        when(resultSet.next()).thenReturn(true).thenReturn(false);
-        when(resultSet.getString("name")).thenReturn(expectedAccount2.getName());
-        when(resultSet.getInt("id")).thenReturn(expectedAccount2.getId());
-        when(resultSet.getFloat("money")).thenReturn(expectedAccount2.getMoney());
-        when(resultSet.getString("iban")).thenReturn(expectedAccount2.getIban());
+        when(resultSet.next()).thenReturn(true).thenReturn(true).thenReturn(false);
+        when(resultSet.getString("name")).thenReturn(expectedAccount1.getName()).thenReturn(expectedAccount2.getName());
+        when(resultSet.getInt("id")).thenReturn(expectedAccount1.getId()).thenReturn(expectedAccount2.getId());
+        when(resultSet.getFloat("money")).thenReturn(expectedAccount1.getMoney()).thenReturn(expectedAccount2.getMoney());
+        when(resultSet.getString("iban")).thenReturn(expectedAccount1.getIban()).thenReturn(expectedAccount2.getIban());
 
         List<Account> full = new ArrayList();
 
         full.add(expectedAccount1);
         full.add(expectedAccount2);
         full = repository.findAll("");
```

Comment:
This is still incorrect. Let's see step by step:
 
* In line `List<Account> full = new ArrayList();` you created an empty list.
* In line 44 and 45 you added two elements into the list, `expectedAccount1`and `expectedAccount2`.
* In line 46 you reassigned the list to the output in the findAll method: `full = repository.findAll("");`, **loosing the two previous elements**. 

Remember in Java (as well as in many other languages), when an object is assigned to another by using the equal operator `a = b;`, object in the left is **pointing to the same memory area than b**. Note this is not making a copy of the element but just pointing to the same allocated memory than b. 

Having said this, what you probably wanted to do is:

```
List<Account> expectedAccounts = new ArrayList();
expectedAccounts.add(expectedAccount1);
expectedAccounts.add(expectedAccount2);

List<Account> full = repository.findAll("");
assertFalse(full.isEmpty());
assertEquals(expectedAccounts, full);
```

Note that the `assertEquals()` is doing a real compare of the objects inside (element by element).

Reply:
Assert equals gives issues when comparing two arrays (I think it compares if they are the same list, rather than comparing if their items are the same).

Deleted the add statements on all three tests, since they are redundant if just comparing an already existing list to a fixed number.

Reply:
`assertEquals` compares the two lists (comparing the elements inside, one by one). The issue here is not in the `assertEquals` itself but our own code. I noticed the `Account` class doesn't implement the `equals` method, which is required when using the JUnit assertions for collections.

I pushed a commit with the `equals()` methods in all the DTOs and the code I suggested to you is working now.